plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/explorer/BaseExplorerContentProvider.java

```diff
@@ -179,13 +179,37 @@ protected IStatus run(IProgressMonitor monitor) {
 		job.schedule();
 		return stub;
 	}
-
-	protected void refreshViewer(final Object object) {
+	
+	private void asyncViewerRefresh(final Runnable runner) {
```

Comment:
Yes I understand but I see the method simply running Runnables which can contain anything you want. When reading "refresh" I would immediately associate it to refreshing the viewer and then discovering in code that's its running general purpose runnables in a asnync manner makes me go "oh, unexpected".
What about #runAsync?